### PR TITLE
Add source files

### PR DESCRIPTION
**Pull request by ghost**
Add source files to LSM6DSL library.
Examples available for Discovery L475VG IOT board.

### CODE REVIEW

**Comment by cparata** [comment]
Hi Fabien,
it is not clear for me the patch on I2C address. The values of the I2C address for LSM6DSL should be 0xD4 if SDO/SA0 pin is connected to ground or 0xD6 if SDO/SA0 pin is connected to supply voltage. In the case of Discovery IoT Node the SDO/SA0 pin is connected to ground, so the correct way to create the LSM6DSL instance in the examples is:

AccGyr = new LSM6DSLSensor(dev_i2c, LSM6DSL_ACC_GYRO_I2C_ADDRESS_LOW);

In the case of X-NUCLEO_IKS01A2 board instead the SDO/SA0 pin is connected to supply voltage and so the correct way to create the LSM6DSL instance in the examples is:

AccGyr = new LSM6DSLSensor(dev_i2c, LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH);
or simply
AccGyr = new LSM6DSLSensor(dev_i2c);

Why did you change the value of LSM6DSL_ACC_GYRO_I2C_ADDRESS_HIGH?
Best Regards,

Carlo

**Comment by ghost** [comment]
Hi @cparata 

You're right. This modification is useless. 
I removed it and I modified the examples to call the low address.

Thanks you.

File: library.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name=LSM6DSL
+version=1.0
+author=Wi6Labs
+maintainer=STMicroelectronics
```
Comment (fpistm) [inline library.properties]: maintainer=stm32duino

File: keywords.txt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+#######################################
+
+Enable_X											KEYWORD2
+Enable_G											KEYWORD2
+Disable_X											KEYWORD2
+Disable_G											KEYWORD2
+ReadID												KEYWORD2
+Get_X_Axes										KEYWORD2
+Get_G_Axes										KEYWORD2
+Get_X_Sensitivity							KEYWORD2
```
Comment (fpistm) [inline keywords.txt]: Please review indent

**Comment by cparata** [comment]
Hi @fprwi6labs, I tested your code and it is fine. I also added the examples for X-NUCLEO boards, so I suggest to directly use my PR that includes your code plus some small cosmetic fixes.

**Comment by ghost** [comment]
:+1:

**Comment by cparata** [comment]
I close this PR because it is already included in the current official code.

**Pull request closed without merging**